internal/core/algorithm/ngt/ngt.go

```diff
@@ -470,14 +476,22 @@ func (n *ngt) LinearSearch(vec []float32, size int) (result []SearchResult, err
 	rsize := int(C.ngt_get_result_size(results, ebuf))
 	if rsize <= 0 {
 		err = n.newGoError(ebuf)
-		if err == nil {
-			err = errors.ErrEmptySearchResult
+		if atomic.LoadUint64(&n.cnt) == 0 {
+			return nil, errors.ErrSearchResultEmptyButNoDataStored
```

Comment:
is there any meaning to check the count after the go error is created from ebuf? 
Can we create the go error after the count is checked?

e.g.
```go
...
if atomic.LoadUint64(&n.cnt) == 0 {
	return nil, errors.ErrSearchResultEmptyButNoDataStored
}

err = n.newGoError(ebuf)
if err != nil {
...
```

Reply:
Usually rsize = 0 is considered an error, which means that the ebuf needs to be returned to the Pool as soon as possible, and therefore, PutErrorBuffer or newGoError can be called to Put or DestroyCreate the ErrorObject, regardless of the type of error or more. 
In the case of ErrorSearchResultEmptyButNoDataStored, there is no need to return the result of newGoError because the error can occur only in the unindexed case, so no Wrap is also performed.